File: coredent-api/app/schemas/reports.py
from pydantic import BaseModel
from typing import List, Dict, Any
from datetime import date, datetime

class AppointmentTypeCount(BaseModel):
    type: str
    count: int
    color: str

class AppointmentDayCount(BaseModel):
    day: str
    count: number

class AppointmentsSummary(BaseModel):
    total: int
    completed: int
    cancelled: int
    noShow: int
    scheduled: int
    completionRate: float
    noShowRate: float
    byType: List[AppointmentTypeCount]
    byDay: List[AppointmentDayCount]

class MonthRevenue(BaseModel):
    month: str
    revenue: float
    collected: float

class ProcedureRevenue(BaseModel):
    procedure: str
    revenue: float
    count: int

class RevenueSummary(BaseModel):
    totalRevenue: float
    totalCollected: float
    totalOutstanding: float
    averagePerVisit: float
    byMonth: List[MonthRevenue]
    byProcedure: List[ProcedureRevenue]

class MonthAcceptance(BaseModel):
    month: str
    proposed: int
    accepted: int

class TreatmentAcceptance(BaseModel):
    proposedPlans: int
    acceptedPlans: int
    completedPlans: int
    acceptanceRate: float
    completionRate: float
    byMonth: List[MonthAcceptance]

class HourUtilization(BaseModel):
    hour: str
    utilization: float

class ChairUtilizationMetric(BaseModel):
    chair: str
    utilization: float
    appointments: int

class DayUtilization(BaseModel):
    day: str
    utilization: float

class ChairUtilizationSummary(BaseModel):
    totalChairs: int
    averageUtilization: float
    peakHours: List[HourUtilization]
    byChair: List[ChairUtilizationMetric]
    byDayOfWeek: List[DayUtilization]

class DashboardMetricsResponse(BaseModel):
    appointments: AppointmentsSummary
    revenue: RevenueSummary
    treatmentAcceptance: TreatmentAcceptance
    chairUtilization: ChairUtilizationSummary

File: coredent-style-main/src/types/reports.ts
// ============================================
// CoreDent PMS - Reports Types
// ============================================

export type ReportType = 'appointments' | 'revenue' | 'treatment' | 'utilization';

export interface DateRange {
  from: Date;
  to: Date;
}

export interface AppointmentTypeMetric {
  type: string;
  count: number;
}

export interface AppointmentDayMetric {
  date: string;
  appointments: number;
}

export interface RevenueMonthMetric {
  month: string;
  revenue: number;
  collected: number;
}

export interface RevenueProcedureMetric {
  procedure: string;
  revenue: number;
}

export interface ChairMetric {
  chair: string;
  utilization: number;
  appointments: number;
}

export interface PeakHourMetric {
  hour: string;
  utilization: number;
}

export interface DayOfWeekMetric {
  day: string;
  utilization: number;
}

export interface DashboardMetrics {
  appointments: {
    total: number;
    completed: number;
    cancelled: number;
    noShow: number;
    completionRate: number;
    noShowRate: number;
    byType: AppointmentTypeMetric[];
    byDay: AppointmentDayMetric[];
  };
  revenue: {
    totalRevenue: number;
    totalCollected: number;
    totalOutstanding: number;
    averagePerVisit: number;
    byMonth: RevenueMonthMetric[];
    byProcedure: RevenueProcedureMetric[];
  };
  treatmentAcceptance: {
    proposedPlans: number;
    acceptedPlans: number;
    completedPlans: number;
    acceptanceRate: number;
    completionRate: number;
  };
  chairUtilization: {
    averageUtilization: number;
    totalChairs: number;
    peakHours: PeakHourMetric[];
    byChair: ChairMetric[];
    byDayOfWeek: DayOfWeekMetric[];
  };
}
